### PR TITLE
Np 3525 authorized access 

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/AuthorizedQueryPersonHandler.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import static no.unit.nva.cristin.common.client.CristinAuthenticator.getHttpClient;
+import static no.unit.nva.cristin.model.Constants.DEFAULT_RESPONSE_MEDIA_TYPES;
+import static no.unit.nva.cristin.model.Constants.PERSON_CONTEXT;
+
+public class AuthorizedQueryPersonHandler extends ApiGatewayHandler<Void, Person> {
+
+    public static final String ERROR_MESSAGE_INVALID_PARAMETER_FOR_PERSON_ID =
+            "Invalid query parameter for national identifier, must be a number with 11 digits";
+    public static final int NATIONAL_IDENTIFIER_LENGTH = 11;
+    private static final String NationalIdentifierNumber = "nin";
```
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/AuthorizedQueryPersonHandler.java:27]: Uppercase if it is a constant? Not important though

File: cristin-person/src/main/java/no/unit/nva/cristin/person/client/AuthorizedCristinPersonApiClient.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import java.net.http.HttpResponse;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static no.unit.nva.cristin.common.ErrorMessages.ERROR_MESSAGE_IDENTIFIER_NOT_FOUND_FOR_URI;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_API_URL;
+import static no.unit.nva.cristin.person.client.CristinPersonQuery.CRISTIN_API_PERSONS_PATH;
+import static no.unit.nva.utils.UriUtils.PERSON;
+
+public class AuthorizedCristinPersonApiClient extends CristinPersonApiClient {
```
Comment (axthosarouris) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/client/AuthorizedCristinPersonApiClient.java:20]: I am not sure whether this client should be called Authorized. The only thing that makes it authorized is the fact that the HttpClient is authorized.

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/AuthorizedQueryPersonHandler.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    @JacocoGenerated
+    public AuthorizedQueryPersonHandler(Environment environment) {
+        super(Void.class, environment);
+    }
+
+    @Override
+    protected Person processInput(Void input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        String identifier = getValidId(requestInfo);
+        Person person = getAuthorizedCristinPersonApiClient().getCristinPerson(identifier).toPerson();
```
Comment (axthosarouris) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/AuthorizedQueryPersonHandler.java:42]: Can't `getAuthorizedCristinPersonApiClient()` be a static constant? that would save some time

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/AuthorizedQueryPersonHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    @Test
+    void shouldReturn200ResponseWhenCallingEndpointWithNationalIdentifierParameter() throws IOException {
+        InputStream input = requestWithQueryParameters(Map.of(NIN, SAMPLE_NIN));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<Person> gatewayResponse = GatewayResponse.fromOutputStream(output);
+        assertEquals(HTTP_OK, gatewayResponse.getStatusCode());
+    }
+
+    @Test
+    void shouldReturn500BadRequestWhenCallingEndpointWithIllegalNationalIdentifierParameter() throws IOException {
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/AuthorizedQueryPersonHandlerTest.java:68]: 500 is not BadRequest.

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by torarnet — COMMENTED** [review]
Looks good but awaiting approval until i get some feedback from team regarding a specific feature

**Pull request closed without merging**